iroh-net/src/netcheck/reportgen.rs

```diff
@@ -249,6 +249,7 @@ impl Actor {
                 break;
             }
             tokio::select! {
+                biased;
```

Comment:
what's your intuition here?

Reply:
Generally every select should be biased, unless there is a compelling reason not too.  Not using biased will result in a random number being generated every time which is a bit of a waste usually.

It would probably have helped here as well, since the failure mode from the timers expiring would be a lot more obvious: what it did now was randomly poll some of the other futures first before finally coming round to "oh, guess we should have stopped".  The log output would have been easier to understand.